src/lib.rs

```diff
@@ -41,17 +41,30 @@ pub use plugin::PluginInitializeReturn;
 use treesitter::{get_matches, maybe_get_query};
 
 #[derive(Parser)]
+#[clap(group(
```

Comment:
The objective here is to convince `clap` that (a) `--query-source` + `--query-file` are mutually exclusive (b) you can't omit all of `--query-source`, `--query-file` and `--filter`

I tried a couple different ways of expressing that (see eg the commented-out lines below) but while this produces some wonky `clap` "help"-style output (see some tests below) it seemed better than the alternatives I was able to figure out